phpBB/phpbb/members/controller/online.php

```diff
@@ -0,0 +1,618 @@
+			'S_SWITCH_GUEST_DISPLAY'	=> ($this->config['load_online_guests']) ? true : false,
+			'S_VIEWONLINE'				=> true,
+		));
+
+		$this->template->assign_block_vars('navlinks', array(
+			'BREADCRUMB_NAME'	=> $this->language->lang('WHO_IS_ONLINE'),
+			'U_BREADCRUMB'		=> $this->helper->route('phpbb_members_online'),
+		));
+
+		make_jumpbox(append_sid("{$phpbb_root_path}viewforum.$phpEx"));
+
+		// Render
+		return $this->helper->render('viewonline_body.html', $this->language->lang('WHO_IS_ONLINE'));
+	}
+
+	protected function get_location($session_page, $forum_id) // $forum_id can be removed in the future i think https://tracker.phpbb.com/browse/PHPBB3-15434
+	{
+		global $phpbb_container, $phpbb_root_path, $phpEx, $phpbb_adm_relative_path;
+
+		try
+		{
+			$session_page = parse_url($session_page,  PHP_URL_PATH); // Remove query string
+			$session_page = '/' . ((substr($session_page, 0, 8) == 'app.php/') ? substr($session_page, 8) : $session_page); // Remove app.php/
+
+			$match = $this->router->match($session_page);
+
+			switch ($match['_route'])
+			{
+				case 'phpbb_help_bbcode_controller':
+				case 'phpbb_help_bbcode_controller':
+					$location = $this->language->lang('VIEWING_FAQ');
+					$location_url = $this->helper->route('phpbb_help_faq_controller');
+				break;
+
+				case 'phpbb_members_online':
+				case 'phpbb_members_online_whois':
+					$location = $this->language->lang('VIEWING_ONLINE');
+					$location_url = $this->helper->route('phpbb_members_online');
+				break;
+
+				case 'phpbb_report_pm_controller':
+				case 'phpbb_report_post_controller':
+					$location = $this->language->lang('REPORTING_POST');
+					$location_url = append_sid("{$phpbb_root_path}index.$phpEx");
+				break;
+
+				case 'phpbb_message_admin':
+				//case 'phpbb_message_topic':
+				//case 'phpbb_message_user':
+					$location = $this->language->lang('VIEWING_CONTACT_ADMIN');
+					$location_url = append_sid("{$phpbb_root_path}memberlist.$phpEx", 'mode=contactadmin');
+				break;
+
+				case 'phpbb_members_profile':
+					$location_url = append_sid("{$phpbb_root_path}memberlist.$phpEx");
+					$location = $this->language->lang('VIEWING_MEMBER_PROFILE');
+				break;
+
+				case 'phpbb_members_team':
+					$location_url = append_sid("{$phpbb_root_path}memberlist.$phpEx");
+					$location = $this->language->lang('VIEWING_MEMBERS');
+				break;
+
+				default:
+					// Is a route, but not in the switch
+					$location = $this->language->lang('INDEX');
+					$location_url = append_sid("{$phpbb_root_path}index.$phpEx");
+			}
+		}
+		catch (\RuntimeException $e) // Urls without route
+		{
+			/* @var $viewonline_helper \phpbb\viewonline_helper */
+			$viewonline_helper = $phpbb_container->get('viewonline_helper');
+			$on_page = $viewonline_helper->get_user_page($session_page);
+
+			switch ($on_page[1])
+			{
+				case 'index':
```

Comment:
This case duplicates default.

Reply:
No, there are diferent things, one is index page, and the other uknown page

Reply:
Looks exactly the same to me: https://github.com/phpbb/phpbb/pull/5410/files#diff-bf89cb421bf9f6fb8770a54ba5711efcR542

Reply:
No, default is for pages that are not in the switch, and index is for index.
The default lang page could be unknown page instead of index for example.